src/lib/formats/dmk_dsk.cpp

```diff
@@ -95,117 +114,207 @@ bool dmk_format::load(util::random_read &io, uint32_t form_factor, const std::ve
 	const int track_size = ( header[3] << 8 ) | header[2];
 	const int heads = (header[4] & 0x10) ? 1 : 2;
 	const bool is_sd = (header[4] & 0x40) ? true : false;
-	//const int raw_track_size = 2 * 8 * ( track_size - 0x80 );
 
+	auto variant = floppy_image::SSDD;
 	if (is_sd)
 	{
 		if (heads == 2)
 		{
-			image->set_variant(floppy_image::DSSD);
+			variant = floppy_image::DSSD;
 		}
 		else
 		{
-			image->set_variant(floppy_image::SSSD);
+			variant = floppy_image::SSSD;
 		}
 	}
 	else
 	{
 		if (heads == 2)
 		{
-			image->set_variant(floppy_image::DSDD);
+			variant = floppy_image::DSDD;
 		}
 		else
 		{
-			image->set_variant(floppy_image::SSDD);
+			variant = floppy_image::SSDD;
 		}
 	}
+	image->set_variant(variant);
+
+	int fm_stride = is_sd ? 1 : 2;
 
 	for (int track = 0; track < tracks; track++)
 	{
 		for (int head = 0; head < heads; head++)
 		{
+			int fm_loss = 0;
 			std::vector<uint8_t> track_data(track_size);
 			std::vector<uint32_t> raw_track_data;
+			int mark_location[64*2+1];
+			uint8_t mark_value[64*2+1];
+			bool mark_is_mfm[64*2+1];
 			int iam_location = -1;
-			int idam_location[64];
-			int dam_location[64];
 
 			// Read track
 			io.read_at(header_size + (heads * track + head) * track_size, &track_data[0], track_size, actual);
 
-			for (int i = 0; i < 64; i++)
+			for (int i = 0; i < 64*2+1; i++)
 			{
-				idam_location[i] = -1;
-				dam_location[i] = -1;
+				mark_location[i] = -1;
+				mark_value[i] = 0xfe;
+				mark_is_mfm[i] = is_sd ? false : true;	// Use default encoding
```

Comment:
`!is_sd` is equivalent to the ternary expression here.

Reply:
Switched to !is_sd